File: examples/csharp_demo/Catalog/Author.cs
namespace Catalog;

public class Author
{
    public string Name { get; set; }
    public string Country { get; set; } = "Unknown";

    public Author(string name)
    {
        Name = name;
    }

    public string DisplayName()
    {
        return $"{Name} ({Country})";
    }
}

File: examples/typescript_demo/src/catalog/author.ts
/** Authors in the bookstore catalog. */
export class Author {
  name: string;
  country: string;

  constructor(name: string, country: string = "Unknown") {
    this.name = name;
    this.country = country;
  }

  displayName(): string {
    return `${this.name} (${this.country})`;
  }
}
